synth/snsynth/transform/onehot.py

```diff
@@ -27,6 +27,9 @@ def _clear_fit(self):
     def _transform(self, val):
         if self.max < 0 or not self._fit_complete:
             raise ValueError("OneHotEncoder has not been fit yet.")
+        elif self.max == 0:
+            return 1
```

Comment:
Can we throw an error here if self.max == 0 and val != 0?

Reply:
By the way: Your CI tests keep failing sporadically. The culprit seems to be `TestFactory.test_fit_with_data_frame()` if the sample of a synthesizer does not contain one row with `married == 0`. I can fix the test if you suggest what it should look like instead.

Reply:
This fix looks great; I'll merge and push a release.  Regarding the CI, I can ignore the failure for now, since it just appears to be catching a very poor fit for the non-CT DPGAN, and I haven't decided what to do about that.  The non-CT GAN-based synthesizers aren't very good, so maybe we deprecate them, or at least stop validating their outputs, or maybe do some more checking to see if they are better with continuous values or something?

Reply:
One other comment: the sklearn OneHotEncoder returns a list of lists even if there is only one category, and there are good reasons to do that.  But as you noticed, we break downstream if there are any tuples of width 1, since we assume that all columns can be nullable, which immediately turns them into a tuple, and so we use output_width == 1 as a way to tell that something should be a value rather than a tuple.  We could rework everything downstream to preserver closer compatibility with sklearn in this edge case, but I think your solution is better.

Reply:
Okay good. Seems like you just closed the PR without merging (by accident?).

> The non-CT GAN-based synthesizers aren't very good, so maybe we deprecate them, or at least stop validating their outputs, or maybe do some more checking to see if they are better with continuous values or something?

Since they do not provide good results for synthetic tabular data, I would second your deprecation proposal. Or at least make it clear to a user that the model is untested and might be an inadequate synthesizer choice.

> We could rework everything downstream to preserver closer compatibility with sklearn in this edge case, but I think your solution is better.

Thank you for providing your insight. I don't think there is a need to rework the internals, at least not currently. A user can use sklearn separately, without a `TableTransformer`.